src/dsp_tools/commands/xmlupload/resource_multimedia.py

```diff
@@ -68,22 +80,49 @@ def handle_bitstream(
         The information from sipi which is needed to establish a link from the resource
     """
     try:
-        if preprocessing_done:
-            resource_bitstream = resource.get_bitstream_information(bitstream.value, permissions_lookup)
-        else:
-            resource_bitstream = _upload_bitstream(
-                resource=resource,
-                sipi_server=sipi_server,
-                imgdir=imgdir,
-                permissions_lookup=permissions_lookup,
-            )
-            msg = f"Uploaded file '{bitstream.value}'"
-            print(f"{datetime.now()}: {msg}")
-            logger.info(msg)
+        resource_bitstream = _upload_bitstream(
+            resource=resource,
+            sipi_server=sipi_server,
+            imgdir=imgdir,
+            permissions_lookup=permissions_lookup,
+        )
+        msg = f"Uploaded file '{bitstream.value}'"
+        print(f"{datetime.now()}: {msg}")
+        logger.info(msg)
         return resource_bitstream
     except BaseError as err:
         err_msg = err.orig_err_msg_from_api or err.message
         msg = f"Unable to upload file '{bitstream.value}' of resource '{resource.label}' ({resource.id})"
         print(f"{datetime.now()}: WARNING: {msg}: {err_msg}")
         logger.warning(msg, exc_info=True)
         return None
+
+
+def _upload_bitstream(
+    resource: XMLResource,
+    sipi_server: Sipi,
+    imgdir: str,
+    permissions_lookup: dict[str, Permissions],
+) -> BitstreamInfo | None:
+    """
+    This function uploads a specified bitstream file to SIPI and then returns the file information from SIPI.
+
+    Args:
+        resource: resource with that has a bitstream
+        sipi_server: server to upload
+        imgdir: directory of the file
+        permissions_lookup: dictionary that contains the permission name as string and the corresponding Python object
+
+    Returns:
+        The information from sipi which is needed to establish a link from the resource
+    """
+    pth = resource.bitstream.value  # type: ignore[union-attr]
```

Comment:
`_upload_bitstream()` is called where the none case for resource.bitstream was already done. If you would just pass `path` from there, then you wouldn't need the type ignore